src/core/variable.jl

```diff
@@ -180,6 +180,272 @@ function variable_mc_branch_flow_reactive(pm::_PMs.AbstractPowerModel; nw::Int=p
+        ) for l in _PMs.ids(pm, nw, :branch)
+    )
+
+    if bounded
+        ub = Dict()
+        for (l,i,j) in _PMs.ref(pm, nw, :arcs_from)
+            b = branch[l]
+            ub[l] = Inf
+            if haskey(b, "rate_a")
+                rate = b["rate_a"].*b["tap"]
+                y_fr = abs.(b["g_fr"] + im*b["b_fr"])
+                y_to = abs.(b["g_to"] + im*b["b_to"])
+                shuntcurrent = max.(y_fr*bus[i]["vmax"].^2, y_to*bus[j]["vmax"].^2)
+                seriescurrent = max.(rate./bus[i]["vmin"], rate./bus[j]["vmin"])
+                ub[l] = seriescurrent + shuntcurrent
+            end
+            if haskey(b, "c_rating_a")
+                totalcurrent = b["c_rating_a"]
+                y_fr = abs.(b["g_fr"] + im*b["b_fr"])
+                y_to = abs.(b["g_to"] + im*b["b_to"])
+                shuntcurrent = max.(y_fr*bus[i]["vmax"].^2, y_to*bus[j]["vmax"].^2)
+                ub[l] = totalcurrent + shuntcurrent
+            end
+        end
+
+        for l in  _PMs.ids(pm, nw, :branch)
+            for c in _PMs.conductor_ids(pm; nw=nw)
+                if !isinf(ub[l][c])
+                    JuMP.set_lower_bound(csi[l][c], -ub[l][c])
+                    JuMP.set_upper_bound(csi[l][c],  ub[l][c])
+                end
+            end
+        end
+    end
+    report && _PMs.sol_component_value(pm, nw, :branch, :csi_fr, _PMs.ids(pm, nw, :branch), csi)
+end
+
+
+"variable: `cr[l,i,j]` for `(l,i,j)` in `arcs`"
+function variable_mc_transformer_current_real(pm::_PMs.AbstractPowerModel; nw::Int=pm.cnw, bounded::Bool=true, report::Bool=true)
+    #trans = _PMs.ref(pm, nw, :transformer)
+    #bus = _PMs.ref(pm, nw, :bus)
+    cnds = _PMs.conductor_ids(pm; nw=nw)
+    ncnds = length(cnds)
+
+    cr = _PMs.var(pm, nw)[:crt] = Dict((l,i,j) => JuMP.@variable(pm.model,
+            [c in 1:ncnds], base_name="$(nw)_crt_$((l,i,j))",
+            start = comp_start_value(_PMs.ref(pm, nw, :transformer, l), "cr_start", c, 0.0)
+        ) for (l,i,j) in _PMs.ref(pm, nw, :arcs_trans)
+    )
+
+    #TODO add bounds
+    # if bounded
+    #     ub = Dict()
+    #     for (l,i,j) in _PMs.ref(pm, nw, :arcs_from)
+    #         b = branch[l]
+    #         # ub[l] = Inf
+    #         if haskey(b, "rate_a")
+    #             rate_fr = b["rate_a"].*b["tap"]
+    #             rate_to = b["rate_a"]
+    #             ub[l]  = max.(rate_fr./bus[i]["vmin"], rate_to./bus[j]["vmin"])
+    #         end
+    #         if haskey(b, "c_rating_a")
+    #             ub[l] = b["c_rating_a"]
+    #         end
+    #     end
+    #
+    #     for (l,i,j) in _PMs.ref(pm, nw, :arcs)
+    #         for c in _PMs.conductor_ids(pm; nw=nw)
+    #             if !isinf(ub[l][c])
+    #                 JuMP.set_lower_bound(cr[(l,i,j)][c], -ub[l][c])
+    #                 JuMP.set_upper_bound(cr[(l,i,j)][c],  ub[l][c])
+    #             end
+    #         end
+    #     end
+    # end
```

Comment:
This should be added but the current data model lacks clear support for transformer current/power bounds. I would add this back in after the data model has been added.

Reply:
same reason as before

Reply:
i think because it just uses rate_a and c_rating_a that we shouldn't comment this out, and we should resolve the correct selection of those values in a future PR

Reply:
Okay, I'll patch it up this afternoon.

Reply:
I followed a pattern similar to the branches for it; I think it is good to go now.